autotx/tests/agents/token/research/test_research_swap_and_send.py

```diff
@@ -0,0 +1,93 @@
+from autotx.utils.ethereum import get_erc20_balance, load_w3
+from autotx.utils.ethereum.eth_address import ETHAddress
+
+DIFFERENCE_PERCENTAGE = 0.01
+
+def test_research_buy_one_send_one(configuration, auto_tx, test_accounts):
+    (_, _, _, manager) = configuration
+    web3 = load_w3()
+
+    receiver = test_accounts[0]
+    
+    shib_address = ETHAddress(auto_tx.network.tokens["shib"])
```

Comment:
we have a lot of tests that fetch the meme coin with largest market cap - since this is dynamic, should we get the data from coingecko in each test?

Reply:
You're right, I just made an issue for it here: https://github.com/polywrap/AutoTx/issues/247
Seems a bit lower prio though since the highest market cap coins won't likely change that soon